h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1008,17 +1013,70 @@ AutoBuffer putUdp(UDP.udp type, int senderPort){
    *
    * @param type type of the UDP datagram
    */
-  AutoBuffer putUdp (UDP.udp type) {
-    return putUdp(type, H2O.H2O_PORT); // Outgoing port is always the sender's (me) port
+  AutoBuffer putUdp(UDP.udp type) {
+    return putUdp(type, H2O.H2O_PORT);
+  }
+
+  /**
+   * Select last 15 bytes from the jvm boot start time and return it as char
+   */
+  public static char createUniqueId(long jvmStartTime){
+    byte availableBits = 15;
+    int bitMask = (1 << availableBits) - 1;
+
+    // we set the lower 15 bits, the 1 bit is currently set to 0 which means we are not a client
+    return (char) (jvmStartTime & bitMask);
+  }
+
+
+  private static char CLIENT_CODE_MASK = (char) (1 << 15);
```

Comment:
we could just consider the sign to be the indicator (negative values clients, non-negative regular members) that would a little bit simplify the math and the decoding functions

Reply:
Yup, will simplify a lot, we can use short for this

Reply:
Implemented